### PR TITLE
search results copy All with sort like tree

### PR DESCRIPTION
**Pull request by cedric05**

### CODE REVIEW

**Comment by cedric05** [comment]
fixes #48928

File: src/vs/workbench/parts/search/browser/searchActions.ts [hunk 1]
```diff
@@ -758,10 +761,26 @@ export const copyMatchCommand: ICommandHandler = (accessor, match: RenderableMat
 	}
 };
 
+export class SearchSorter implements ISorter {
```
Comment (cedric05) [inline src/vs/workbench/parts/search/browser/searchActions.ts]: direct import is creating circular dependency.
so moved here

File: src/vs/workbench/parts/search/browser/searchActions.ts [hunk 2]
```diff
@@ -717,7 +718,6 @@ function fileMatchToString(fileMatch: FileMatch, maxMatches: number): { text: st
 		.slice(0, maxMatches)
```
Comment (roblourens) [inline src/vs/workbench/parts/search/browser/searchActions.ts]: Do the FileMatches need to be sorted too?

Comment (roblourens) [reply src/vs/workbench/parts/search/browser/searchActions.ts]: I disagree. All matches should show up in the same order that they show up in the search tree.

I think right now they will be sorted by the backend but that's not guaranteed in the future, so we should use the sorter for everything now.

Comment (cedric05) [reply src/vs/workbench/parts/search/browser/searchActions.ts]: To begin with, what is criteria of sorting matches in a file? if you take by line no, then no need of sorting.
SearchTree matches are sorted by filename not by line contents or line number.
I verified SearchTree matches and copy ALL. Both are same to same.

Comment (roblourens) [reply src/vs/workbench/parts/search/browser/searchActions.ts]: They should be sorted by line, which is what the sorter already does. All you have to do is use it the same way as the change you already made.

And the same should be done for folders.

> I verified SearchTree matches and copy ALL. Both are same to same with mychange.

That might happen but it's not guaranteed to happen every time, so we should enforce the sort order here.

**Review by roblourens — APPROVED** [review]
Thanks!

**Pull request merged**